Extend/Controller/OrderController.php

```diff
@@ -128,8 +162,35 @@ public function execute()
             $this->_handleTransaction($oBasket, $oOrder);
         }
 
-        //redirect happens before the return
-        return '';
+        $iSuccess = $oOrder->oxorder__wdoxidee_finalizeorderstate->value;
+        return $this->_getNextStep($iSuccess);
+    }
+
+    /**
+     * @param Session $oSession
+     * @param Payment $oPayment
+     * @param Order   $oOrder
+     *
+     * @throws \Exception
+     */
+    private function _handleFormResponse($oSession, $oPayment, $oOrder)
+    {
+        if (!empty($oResponse = $oSession->getVariable(self::FORM_POST_VARIABLE))) {
```

Comment:
I know OXID does this as well, but I don't think we should do variable assignments in `if` statements. It's generally considered an anti-pattern.

Reply:
👍